majifix311/src/main/java/com/example/majifix311/ui/auth/SigninActivity.java

```diff
@@ -0,0 +1,422 @@
+package com.example.majifix311.ui.auth;
+
+import android.app.ProgressDialog;
+import android.content.Context;
+import android.graphics.drawable.Drawable;
+import android.os.Build;
+import android.os.Bundle;
+import android.support.design.widget.Snackbar;
+import android.support.design.widget.TextInputEditText;
+import android.support.design.widget.TextInputLayout;
+import android.support.v4.content.ContextCompat;
+import android.support.v7.app.AppCompatActivity;
+import android.text.TextUtils;
+import android.util.Patterns;
+import android.view.View;
+import android.view.inputmethod.InputMethodManager;
+import android.widget.Button;
+
+import com.example.majifix311.BuildConfig;
+import com.example.majifix311.R;
+import com.example.majifix311.auth.Auth;
+import com.example.majifix311.models.Party;
+import com.jakewharton.rxbinding2.view.RxView;
+import com.jakewharton.rxbinding2.widget.RxTextView;
+
+import java.net.UnknownHostException;
+import java.util.concurrent.TimeUnit;
+
+import io.reactivex.Observable;
+import io.reactivex.android.schedulers.AndroidSchedulers;
+import io.reactivex.disposables.CompositeDisposable;
+import io.reactivex.disposables.Disposable;
+import io.reactivex.functions.BiFunction;
+import io.reactivex.functions.Consumer;
+import io.reactivex.functions.Predicate;
+import io.reactivex.observers.DisposableObserver;
+import io.reactivex.schedulers.Schedulers;
+import retrofit2.adapter.rxjava2.HttpException;
+
+/**
+ * Reactive Signin activity
+ *
+ * @see <a href="https://academy.realm.io/posts/donn-felker-reactive-android-ui-programming-with-rxbinding/">https://academy.realm.io/posts/donn-felker-reactive-android-ui-programming-with-rxbinding/</a>
+ * @see <a href="https://medium.com/@etiennelawlor/rxjava-on-the-sign-in-screen-9ecb66b88572">https://medium.com/@etiennelawlor/rxjava-on-the-sign-in-screen-9ecb66b88572</a>
+ */
+public class SigninActivity extends AppCompatActivity {
+
+    public static final String TAG = SigninActivity.class.getSimpleName();
+
+    //view references
+    private TextInputLayout emailInputLayout;
+    private TextInputLayout passwordInputLayout;
+    private TextInputEditText emailTextInput;
+    private TextInputEditText passwordTextInput;
+    private Button signInButton;
```

Comment:
I've been, so far, using the initials of the type, followed by its purpose so: `til_email` as id and `mTilEmail` or `tilEmail` as variable name. These names are clear and so I'm not too worried, but at some point we should discuss and decide for consistently.

Reply:
Sure...we should pick a convection here...on `layouts` it make sense to have layout initials(ti, til etc) `prefixing ids` and in codes normally i prefer to favour code readability e.g pronouncing `tilEmail` is different that `emailTextInputLayout` unless your write down the rules. But sure we can resonate on this.